### ISSUE TITLE
Cleanup: Replace all hardcoded <img> tags with $tp->toImage();

### ISSUE THREAD

**Issue by CaMer0n** [e107inc/e107#1342]
<img> tags are inconsistent throughout the core.
For icons we already have toIcon() but for everything else they are currently hard-coded.
Having a central method for rendering will provide enhancements for rendering on hi-res mobile devices/tablets as well as easier updates. (eg. 'img-responsive' will become 'img-fluid' in Bootstrap 4).

I understand this is a HUGE task, so pull-requests are VERY welcome. Just focus on one plugin/area at a time to avoid merge conflicts and test every change to ensure the image is not broken or lost some functionality. 

@Moc @lonalore @tgtje @Jimmi08

**Comment by Jimmi08** [comment]
I take Gallery plugin. Did you mean this?
![image](https://cloud.githubusercontent.com/assets/5429548/12928276/cc51fdfc-cf6c-11e5-84e2-d4d6cd3a1ecf.png)

**Comment by CaMer0n** [comment]
@Jimmi08 Yes, something like that. But $att needs to be merged with array('alt'=>'whatever');

**Comment by Jimmi08** [comment]
@CaMer0n 
I did it by e_rss.php. I don't see diffs. What I am missing?
![image](https://cloud.githubusercontent.com/assets/5429548/12928962/b6a05216-cf70-11e5-9d13-7f2ac354c68f.png)

**Comment by CaMer0n** [comment]
gallery parses some special parms to thumbUrl(), which are in $att. So you need to build on that array by adding alt and class.

**Comment by Jimmi08** [comment]
Yes, I know what you mean. There are reasons why I can't do it and I let it to you  (I can be wrong, of course, I wrote this just to show that I tried :) )
- paths in gallery images (default from installation) includes {e_PLUGIN}, but thumbUrl() is called only for path with e_THEME or e_MEDIA, so they never be hashed with toImage();

-$att is not array in gallery plugin 
![image](https://cloud.githubusercontent.com/assets/5429548/12930334/ecc06e56-cf77-11e5-8af1-356658b7a4c5.png)
and toImage() needs width and height as separate array fields

- toImage() doesn't use x parameter (to be hashed path) . ThumbUrl() is called with options=null, others options are setted by f.e.  $tp->thumbWidth(), but to get hashed path, option x is still needed to be set somehow. 

When somebody other will do this, I will see where I am wrong. 

Just notice: with using img-responsive class with image to get this image centered, you need to use:  'style'=>'margin: 0 auto'  as inline style for toImage().

**Comment by CaMer0n** [comment]
There were some bugs to fix first. Thank you for trying.

**Comment by Jimmi08** [comment]
@CaMer0n Could you check if this is intention?  a tag and img tag have the same class. It's weird to have img-responsive class with a tag. Thanks.
![image](https://cloud.githubusercontent.com/assets/5429548/13362342/11f3090a-dcc4-11e5-8b54-cb19c721ad26.png)
![image](https://cloud.githubusercontent.com/assets/5429548/13362364/3a12c07e-dcc4-11e5-934b-0a735cb4f8ee.png)

**Comment by CaMer0n** [comment]
Normally img-responsive should be on the `<img>` tag. If you remove it from `<a>` how does it look?

**Comment by Jimmi08** [comment]
Good, without change. 
But I notice next problem with description. There should be test if it's empty and this (when prefs downloaded is set to True) look weird too)
![image](https://cloud.githubusercontent.com/assets/5429548/13366697/55f86942-dcdd-11e5-95e4-e6626a2b93d2.png)

**Comment by Jimmi08** [comment]
If this is right solution, I could try to pull request:
![image](https://cloud.githubusercontent.com/assets/5429548/13367024/7fee6362-dcdf-11e5-9288-e27bbe04313f.png)

![image](https://cloud.githubusercontent.com/assets/5429548/13366965/efc83682-dcde-11e5-8a6a-ed8974072f22.png)
Remove description is important, it should be done via template, there is shortcode for this.
